### ISSUE TITLE
Client crash when rendering PokeDex through JEI

### ISSUE THREAD

**Issue by Farsinuce** [Thutmose/Pokecube#210]
#### Issue Description:
Client crashes with latest version of [Just Enough Items](https://minecraft.curseforge.com/projects/just-enough-items-jei/files) at "rendering ingredient pokedex".
Only affected version is `jei_1.12-4.7.5.85.jar` - but not `jei_1.12-4.7.5.83.jar` and below **(UPDATE: see next comment)**.

I am posting this here, since I am guessing that this might be a PokeCube related issue rather than JEI.

#### Steps to reproduce:
1. Open inventory in-game (E)
2. Client crashes

____
#### Affected Versions:
- Pokecube Core-6.20.1.jar
- Pokecube Revival-6.11.1.jar
- Pokecube Mobs-2.14.7.jar
- Minecraft: 1.12
- Forge: 14.21.1.2443
- jei_1.12-4.7.5.85.jar
- pokeplayer-1.12-1.0.0.jar
- pokecubeserverutils-1.5.1.jar

[crash-2017-08-11_17.25.34-client.txt](https://github.com/Thutmose/Pokecube/files/1218348/crash-2017-08-11_17.25.34-client.txt)

**Comment by Farsinuce** [comment]
**UPDATE:**
I experience the bug in `jei_1.12-4.7.5.83.jar` as well after all.
The issue can be replicated by opening your inventory and searching for "gene". Then the client crashes.

**Comment by Thutmose** [comment]
Hmm, somehow it is adding a non-existant pokemob to the list to render. I will get this fixed asap.

**Comment by Thutmose** [comment]
the latest version (6.11.2) should instead print out a warning message instead of crashing, can you let me know which pokemob it seems to be missing? I was unable to replicate this crash on my end.

**Comment by Farsinuce** [comment]
Got it.
`[01:38:24] [main/INFO] [STDERR/]: [pokecube.compat.jei.ingredients.PokedexEntryIngredientRenderer:render:49]: Error rendering Genesect`

**Comment by Thutmose** [comment]
this is very odd, it isn't happening for me:

![image](https://user-images.githubusercontent.com/2363472/29235926-d6286972-7ed1-11e7-9ac0-ad75ce38e130.png)

this is with:
Pokecube Core-6.21.0.jar
Pokecube Revival-6.11.2.jar
Pokecube Mobs-2.14.7.jar
Minecraft: 1.12
Forge: 14.21.1.2443
jei_1.12-4.7.5.85.jar
pokeplayer-1.12-1.0.0.jar
pokecubeserverutils-1.5.1.jar

Do you have any other mods on server/client?

**Comment by Farsinuce** [comment]
These are my installed mods:
```
[1.12_1.12.1] GlobalXP v1.3.jar
AppleSkin-mc1.12-1.0.7.jar
AutoRegLib-1.3-13.jar
BetterAdvancements-1.12-0.0.1.8.jar
Chameleon-1.12-4.1.3.jar
CookingForBlockheads_1.12-6.3.2.jar
gravestone-1.8.4.jar
Hwyla-1.8.20-B35_1.12.jar
jei_1.12-4.7.5.83.jar
journeymap-1.12-5.4.9.jar
JustEnoughResources-1.12-0.8.3.23.jar
LLOverlayReloaded-1.1.4-mc1.12.jar
LunatriusCore-1.12-1.2.0.40-universal.jar
Pam's HarvestCraft 1.12c.jar
Pokecube Core-6.20.1.jar (just reverted from 6.21.0)
Pokecube Mobs-2.14.7.jar
Pokecube Revival-6.11.1.jar (just reverted from 6.11.2)
pokecube_alternative-2.2.3.jar
pokecubeserverutils-1.5.1.jar
pokeplayer-1.12-1.0.0.jar
Quark-r1.2-104.jar
Ruins-1.12.jar
StorageDrawers-1.12-5.2.9.jar
thut_bling-2.0.1.jar
thut_wearables-3.1.0.jar
thutcore-5.11.0.jar
thutessentials-2.1.3.jar
thutpermissions-2.0.0.jar
thuttech-6.0.4.jar
```
The only other modification that I can think of, is that I have deleted generation 3-7 from the Pokecube Mobs jar-file's asset folder.
Then I had to type `/fml confirm` to start my server initially. But I don't know if the modified jar-file is relevant to this JEI/Genesect issue.

**Comment by Thutmose** [comment]
that is the problem then, can you confirm that they were deleted from both the client and server jars?

**Comment by Farsinuce** [comment]
Yes, I can confirm that only generation 1 and 2 remain in client and server jar:
![2017-08-13-000110](https://user-images.githubusercontent.com/5788348/29244868-8de77c80-7fc5-11e7-9655-3e34d28c455d.png)

Edit:
But I have been doing this trick for approx. two weeks (removing poke gen assets from the .jar), and only now, in the new JEI version (and PokeCube Mobs?) the issue is occuring.

**Comment by Thutmose** [comment]
hmm, I wont have pc access for a couple days, I can look into this more when i return, it seems somehow JEI is remembering that genesect was there, even when removed...  can you confirm if this issue still occurs in singleplayer? and also on a brand new world in single player (ie made with a version of mobs with genesect removed)

**Comment by Farsinuce** [comment]
Alright. Have a good couple of days off :)

I can confirm that the issue still occurs in single player in a newly generated world.
Same version of mobs jar (only gen 1 and 2).

Here is the crash report:
[crash-2017-08-13_17.32.31-client.txt](https://github.com/Thutmose/Pokecube/files/1220545/crash-2017-08-13_17.32.31-client.txt)

EDIT:
Is this an isolated issue? I mean; am I and the players on my server the only ones experiencing this issue, hence you are unable to replicate it?
Sorry for taking your time.

**Comment by Farsinuce** [comment]
I have looked through my files in `%appdata%/.minecraft` by random, and found this in pokeadvrecipes.xml:

Line 3: `<Recipe handler="dna" lvl="15" cost="10000" pokemonE="Genesect" chance="0.1" priority="-1" remain="0">//Mew DNA`

Line 29: `<Recipe handler="cloner" lvl="70" cost="30000" pokemon="Genesect">//Genesect`

[It's a long shot](http://i0.kym-cdn.com/photos/images/original/000/234/739/fa5.jpg), but could this be relevant?

**Comment by Thutmose** [comment]
ahh I bet they are relevant, try removing those entries from the list, and then restarting, you might also need to set "force recipes" false in the configs, and maybe also try force database as well.

**Comment by Sinuce** [comment]
Cool. Thanks for the tip.
I'll come back to you in ~17 hours. #timezones

**Comment by Farsinuce** [comment]
Darn. It didn't work.
Here is a copy of the crash-report and the modified xml-file. I also tried to set "force recipes" to false in the PokeCube config.
[txt_xml.zip](https://github.com/Thutmose/Pokecube/files/1226524/txt_xml.zip)

FYI:
I have not modified the xml file or PokeCube config on the server - only in my client, since it is my client that is crashing.

**Comment by Thutmose** [comment]
Try also editing the server stuff,  I can probably look into it on Thursday evening, I am away from desktop till them

**Comment by Farsinuce** [comment]
I will come back to this in a couple of days - after the weekend.

**Comment by Farsinuce** [comment]
Instead of editing the server stuff, I have updated several of the server's mods.
Incl. jei_1.12.1-4.7.7.90.jar

I have posted my list of mods here: https://pastebin.com/P4Tb1M8b

**TL;DR:**
Mods updated + client XML-file edited as mentioned above = Genesect no longer crashes the client.

**Update:**
Other players (without the edited xml-file) are reporting, that a JEI search on Genesect does not crash their game any longer.
It seems like the newest version of JEI has fixed this? Closing issue :)

![2017-08-26-000111](https://user-images.githubusercontent.com/5788348/29735210-8eb6a3de-89f7-11e7-9d2c-2d040dc6e8e6.png)
